core/src/main/java/org/neo4j/ogm/metadata/DomainInfo.java

```diff
@@ -396,96 +384,50 @@ private void registerDefaultFieldConverters(ClassInfo classInfo, FieldInfo field
         }
     }
 
-    private void setEnumFieldConverter(FieldInfo fieldInfo, Class enumClass) {
-        if (fieldInfo.isArray()) {
-            fieldInfo.setPropertyConverter(ConvertibleTypes.getEnumArrayConverter(enumClass));
-        } else if (fieldInfo.isIterable()) {
-            fieldInfo.setPropertyConverter(
-                ConvertibleTypes.getEnumCollectionConverter(enumClass, fieldInfo.getCollectionClassname()));
-        } else {
-            fieldInfo.setPropertyConverter(ConvertibleTypes.getEnumConverter(enumClass));
-        }
+    // leaky for spring
+    public Map<String, ClassInfo> getClassInfoMap() {
+        return classNameToClassInfo;
     }
 
-    private void setBigDecimalConverter(FieldInfo fieldInfo) {
-        if (fieldInfo.isArray()) {
-            fieldInfo.setPropertyConverter(ConvertibleTypes.getBigDecimalArrayConverter());
-        } else if (fieldInfo.isIterable()) {
-            fieldInfo.setPropertyConverter(
-                ConvertibleTypes.getBigDecimalCollectionConverter(fieldInfo.getCollectionClassname()));
-        } else {
-            fieldInfo.setPropertyConverter(ConvertibleTypes.getBigDecimalConverter());
-        }
+    public List<ClassInfo> getClassInfos(String interfaceName) {
+        return interfaceNameToClassInfo.get(interfaceName);
     }
 
-    private void setBigIntegerFieldConverter(FieldInfo fieldInfo) {
-        if (fieldInfo.isArray()) {
-            fieldInfo.setPropertyConverter(ConvertibleTypes.getBigIntegerArrayConverter());
-        } else if (fieldInfo.isIterable()) {
-            fieldInfo.setPropertyConverter(
-                ConvertibleTypes.getBigIntegerCollectionConverter(fieldInfo.getCollectionClassname()));
+    /**
+     * Selects the specialized attribute converter for the given field info, depending wether the field info
+     * describes an array, iterable or scalar value.
+     *
+     * @param fieldInfo must not be {@literal null}.
+     * @param from      The attribute converters to select from, must not be {@literal null}.
+     * @return
+     */
+    private static Optional<AttributeConverter<?, ?>> selectAttributeConverterFor(FieldInfo fieldInfo,
+        AttributeConverters from) {
+
+        FieldInfo fi = Objects.requireNonNull(fieldInfo, "Need a field info");
```

Comment:
Would like to have more meaningful variables name if possible e.g. enough space to write `fieldInfo` here.